src/main.rs

```diff
@@ -62,3 +67,7 @@ pub fn main() {
         server::run_server(analysis, vfs, build_queue);
     }
 }
+
+fn version() -> &'static str {
+    concat!(env!("CARGO_PKG_VERSION"), include_str!(concat!(env!("OUT_DIR"), "/commit-info.txt")))
```

Comment:
Hmm, we could just open the file and read it rather than using include_str, but this is certainly nicer. I guess we can wait and see if someone breaks